scripts/build.sh

```diff
@@ -1,5 +1,22 @@
 #!/bin/bash
-pushd collector/cmd/dcos-metrics-collector
-make
-popd
+set -e
+export CGO_ENABLED=0
 
+COMPONENT="$1"
+PLATFORM=$(uname | tr [:upper:] [:lower:])
+GIT_REF=$(git describe --always)
+SOURCE_DIR=$(git rev-parse --show-toplevel)
+BUILD_DIR="${SOURCE_DIR}/build/${COMPONENT}"
+
+# Bring in dependencies required by schema/go/generator.go
+go get -u github.com/antonholmquist/jason
+
+rm -rf $BUILD_DIR && mkdir -p $BUILD_DIR
+cd "${SOURCE_DIR}/${COMPONENT}"
+
+# build metrics-schema package
+go generate
+
+# build binary
+cd cmd/dcos-metrics-collector
+go build -a -ldflags "-w" -o ${BUILD_DIR}/dcos-metrics-collector-${GIT_REF} *.go
```

Comment:
My thoughts were to omit DWARF symbols (`-w`) for production use because, more often than not, we _probably_ don't need to debug this service after it's been built. I figured if worse comes to worse and we're trying to debug something in a live environment that we can't get through traditional troubleshooting, we can release a `-debug` build for that specific case. If you know of a reason that we might want to keep these around, I'm happy to hear the counterpoint!

This was inspired by https://github.com/golang/go/issues/6853#issuecomment-66088650 (and other various blog posts on the internet).